### PR TITLE
Fix the meta archive-size response.

### PR DESCRIPTION
**Pull request by frankban**
Now the size is wrapped in a struct as required by the spec.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/89/

**Comment by jrwren** [comment]
:+1:

File: internal/v4/api_test.go [hunk 1]
```diff
@@ -107,14 +107,14 @@ var metaEndpoints = []metaEndpoint{{
 	},
 }, {
 	name:            "archive-size", // Charm size.
-	get:             entityFieldGetter("Size"),
+	get:             entityGetter(entitySizeGetter),
 	checkURL:        "cs:precise/wordpress-23",
-	assertCheckData: sizeChecker,
+	assertCheckData: entitySizeChecker,
 }, {
 	name:            "archive-size", // Bundle size.
```
Comment (rogpeppe) [inline internal/v4/api_test.go]: I don't think we need this entry. The endpoint is the same whether the
id is a charm or a bundle. We check both charms and bundles against
this endpoint in the tests.

Once we delete the unneeded endpoint, I'd fold entitySizeGetter
back into the assertCheckData field as a literal function.

**Comment by rogpeppe** [comment]
LGTM with the above change, thanks.

File: internal/v4/api_test.go (hunk 1 above)
Comment (frankban) [inline internal/v4/api_test.go]: Done.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/90/

**Pull request merged**